src/diffusers/loaders/single_file.py

```diff
@@ -247,71 +230,142 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
         local_files_only = kwargs.pop("local_files_only", False)
         revision = kwargs.pop("revision", None)
         torch_dtype = kwargs.pop("torch_dtype", None)
+        load_connected_pipeline = kwargs.pop("load_connected_pipeline", False)
+        local_dir = kwargs.pop("local_dir", None)
+        local_dir_use_symlinks = kwargs.pop("local_dir_use_symlinks", "auto")
 
-        class_name = cls.__name__
-
-        original_config, checkpoint = fetch_ldm_config_and_checkpoint(
-            pretrained_model_link_or_path=pretrained_model_link_or_path,
-            class_name=class_name,
-            original_config_file=original_config_file,
+        checkpoint = load_single_file_model_checkpoint(
```

Comment:
Since `load_single_file_model_checkpoint` supports both model and pipeline level checkpoints, I think `load_single_file_checkpoint()` is a better name.